common/app/common/ApplicationMode.scala

```diff
@@ -0,0 +1,19 @@
+package common
+
+import play.api.{Application, GlobalSettings}
+import play.api.Mode._
+
+object ApplicationMode {
+  private var theMode = Prod
```

Comment:
I just preferred no default value because it shouldn't be used anyway.  e.g. if onStart doesn't get called, local dev will be running in prod mode and will start sending metrics to cloudwatch.  I'm not sure "Prod" could be classed as defensive.